presto-main/src/main/java/io/prestosql/operator/scalar/ArrayContainsSubarray.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.function.Convention;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.OperatorDependency;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlNullable;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.function.TypeParameter;
+import io.prestosql.spi.type.StandardTypes;
+import io.prestosql.spi.type.Type;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.prestosql.spi.function.InvocationConvention.InvocationArgumentConvention.BLOCK_POSITION;
+import static io.prestosql.spi.function.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static io.prestosql.spi.function.InvocationConvention.InvocationReturnConvention.NULLABLE_RETURN;
+import static io.prestosql.spi.function.OperatorType.EQUAL;
+
+@Description("Determines whether given values exist in the array")
+@ScalarFunction("contains_subarray")
+public final class ArrayContainsSubarray
+{
+    private ArrayContainsSubarray() {}
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    @SqlNullable
+    public static Boolean contains(
+            @TypeParameter("T") Type elementType,
+            @OperatorDependency(
+                    operator = EQUAL,
```

Comment:
This should probable use IS_DISTINCT_FROM for comparing elements.

Reply:
IS_DISTINCT_FROM can do this too, but it seems doesn't bring any benefits. Using EQUAL can take advantage of `ArrayContains.contains`, how do you think? @findepi

Reply:
The difference between EQUALS and IS_DISTINCT_FROM is in semantics.

For example, does
```
contains_subarray(ARRAY[1,2,3,NULL], ARRAY[3, NULL])
```
?

furthermore, with nested types, does
```
contains_subarray(ARRAY[ARRAY[1], ARRAY[NULL]], ARRAY[ARRAY[NULL]])
```
?

Reply:
Anyway, this falls under "syntax-needs-review", so @tinkerrrr i would recommend to have agreement on the semantics and syntax before investing more, to avoid back and forth with the code.

Reply:
Got it, is there any reference about the semantics and syntax?

Reply:
the difference between EQUALS (`=`) and `IS NOT DISTINCT FROM` is governed by the SQL spec i believe.
see our docs https://prestosql.io/docs/current/functions/comparison.html#is-distinct-from-and-is-not-distinct-from